tests/platform_tests/test_reboot.py

```diff
@@ -66,7 +66,19 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list, reboot_type=REB
     logging.info("Append the latest reboot type to the queue")
     REBOOT_TYPE_HISTOYR_QUEUE.append(reboot_type)
 
-    check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type)
+    try:
+        check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type)
+    except Exception as e:
+        logging.error("Failing due to {}".format(e))
+        raise e
+    finally:
+        # If supervisor node is rebooted in chassis, linecards also will reboot.
+        # Check if all linecards are back up.
+        if dut.is_supervisor_node():
+            for host in duthosts:
+                if host != dut:
+                    logging.info("checking if {} critical services are up".format(host.hostname))
+                    wait_critical_processes(host)
```

Comment:
We should also check here that all the interfaces are up. wait_critical_processes I believe only checks if the services have started, but port operationally being up takes longer - specifically as switch create on BRCM J2C+ takes much longer time.

Reply:
@vperumal can you check on this ?

Reply:
@vperumal please see this if this PR is needed

Reply:
@sanmalho-git  and @abdosi  - Added the check